### ISSUE TITLE
Cannot search Chinese word inside a sentence in the chat history.

### ISSUE THREAD

**Issue by jjaychen1e** [TelegramMessenger/Telegram-iOS#1036]
### Checklist
- [x] I am reporting an issue in existing functionality that does not work as intended
- [x] I've searched for existing [GitHub issues](https://github.com/telegrammessenger/Telegram-iOS/issues)

### Description
In English, words are segmented by spaces, but in other languages such as Chinese and Japanese, they don't use spaces or other separators to segment words in a sentence. It seems Telegram can only search the entire word surrounded by some separators, it's very inconvenient and even unusable for Chinese and Japanese users.

### Expected Behavior
When search "今天" for "今天天气真好"，Telegram will show "No Result".

### Actual Behavior
When search "今天" for "今天天气真好"，Telegram will show "今天天气真好".

### Steps to Reproduce
1. Send "今天天气真好" to saved messages
2. Search "今天" in saved messages
3.  The result is: "No Result"

### Environment
**Device:** `iPhone 14 Pro`

**iOS version**: `16`

**App version:** `9.4.3`

**Comment by DavidLeeMr** [comment]
Yes, I also found the same problem, hope to support Chinese search

**Comment by nawta** [comment]
I found the same problem in Japanese too.
